### ISSUE TITLE
"<unknown> doesn't implement Display" false positive

### ISSUE THREAD

**Issue by lancelote** [intellij-rust/intellij-rust#5378]
## Environment

* **IntelliJ Rust plugin version:** 0.2.123.SNAPSHOT-201-dev
* **Rust toolchain version:** 1.43.0 (4fb7144ed 2020-04-20) x86_64-apple-darwin
* **IDE name and version:** CLion 2020.1.1 (CL-201.7223.86)
* **Operating system:** macOS 10.15.4

## Problem description

The plugin can't infer type after `par_iter` call (old problem with `rayon`) and complains about unimplemented `Display` (new problem). Could the error be suppressed for such cases?

<img width="659" alt="error" src="https://user-images.githubusercontent.com/2500320/81978623-785c4480-9634-11ea-8e1d-9016234d9870.png">

This is a recent regression, as 0.2.122.3078-201-nightly doesn't complain.

## Steps to reproduce

```toml
rayon = "1.3.0"
```

```rust
use rayon::prelude::*;
use std::thread::sleep;
use std::time::Duration;

fn main() {
    let mut v = Vec::with_capacity(500);
    for i in 0..500 {
        v.push(i);
    }

    let v2: Vec<i32> = (&v)
        .par_iter()
        .map(|&x| {
            sleep(Duration::from_millis(100));
            println!("{}", x);
            x * x
        })
        .collect();

    println!("{:?}", &v2[490..500]);
}
```
